typing_extensions/src_py3/test_typing_extensions.py

```diff
@@ -0,0 +1,636 @@
+import sys
+import os
+import abc
+import contextlib
+import collections
+from unittest import TestCase, main, skipUnless
+from typing import TypeVar, Optional
+from typing import T, KT, VT  # Not in __all__.
+from typing import Tuple, List
+from typing import Generic
+from typing import get_type_hints
+from typing import no_type_check
+from typing_extensions import NoReturn, ClassVar, Type, NewType
+import typing
+import typing_extensions
+try:
+    import collections.abc as collections_abc
+except ImportError:
+    import collections as collections_abc  # Fallback for PY3.2.
+
+TYPING_V2 = sys.version_info >= (3, 5, 1)
+TYPING_V3 = sys.version_info >= (3, 5, 3)
+TYPING_V4 = sys.version_info >= (3, 6, 1)
+
+
+class BaseTestCase(TestCase):
+    def assertIsSubclass(self, cls, class_or_tuple, msg=None):
+        if not issubclass(cls, class_or_tuple):
+            message = '%r is not a subclass of %r' % (cls, class_or_tuple)
+            if msg is not None:
+                message += ' : %s' % msg
+            raise self.failureException(message)
+
+    def assertNotIsSubclass(self, cls, class_or_tuple, msg=None):
+        if issubclass(cls, class_or_tuple):
+            message = '%r is a subclass of %r' % (cls, class_or_tuple)
+            if msg is not None:
+                message += ' : %s' % msg
+            raise self.failureException(message)
+
+    def clear_caches(self):
+        for f in typing._cleanups:
+            f()
+
+
+class Employee:
+    pass
+
+
+class NoReturnTests(BaseTestCase):
+
+    def test_noreturn_instance_type_error(self):
+        with self.assertRaises(TypeError):
+            isinstance(42, NoReturn)
+
+    def test_noreturn_subclass_type_error_1(self):
+        with self.assertRaises(TypeError):
+            issubclass(Employee, NoReturn)
+
+    @skipUnless(TYPING_V3, "Behavior added in typing v3")
```

Comment:
It is not necessary, but it would be super-helpful to mention PR numbers where behavior was changed (also for other changes).